File: python/algorithms/InsertionSort/insertion_sort.py
def asc_insertion_sort(arr):
    length = len(arr)
    i = 1
    while (i < length):
        sentry = arr[i]
        j = i - 1
        while (j >= 0) & (arr[j] > sentry):
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = sentry
        i += 1

    return arr


def des_insertion_sort(arr):
    length = len(arr)
    i = 1
    while (i < length):
        sentry = arr[i]
        j = i - 1
        while (j >= 0) & (arr[j] < sentry):
            arr[j+1] = arr[j]
            j -= 1
        arr[j+1] = sentry
        i += 1

    return arr


if __name__ == '__main__':
    main()

File: c/Algorithms/Sort/InsertionSort/insertion_sort.c
/**********************************************
 > File Name: insertion_sort.c
 > Author: Signal Yu
 > Mail: signalfish38861721@gmail.com
 > Time: 11/13/2022
 > Desc:
***********************************************/

#include "insertion_sort.h"
#include <stdio.h>

void insertion_sort(int *arr, int size) {
  for (int i = 1; i < size; i++) {
    int sentry = arr[i];
    int j;
    for (j = i - 1; j >= 0 && arr[j] > sentry; j--) {
      arr[j + 1] = arr[j];
    }
    arr[j + 1] = sentry;
  }
}